R/time_interpolate.R

```diff
@@ -18,81 +18,90 @@
+    sortedYears                <-  sort(c(dimnames(dataset)[[2]], single_interpolated_year))
+    if (sortedYears[1] == single_interpolated_year) {
+      yearBefore <- sortedYears[2]
+      yearAfter  <- sortedYears[3]
+      yearExtrapolate <- ifelse(extrapolation_type == "constant", sortedYears[2], -1)
+    } else if (sortedYears[length(sortedYears)] == single_interpolated_year) {
+      yearBefore <- sortedYears[length(sortedYears) - 2]
+      yearAfter  <- sortedYears[length(sortedYears) - 1]
+      yearExtrapolate <- ifelse(extrapolation_type == "constant", sortedYears[length(sortedYears) - 1], -1)
+    } else {
+      yearBefore <- sortedYears[which(sortedYears == single_interpolated_year) - 1]
+      yearAfter <- sortedYears[which(sortedYears == single_interpolated_year) + 1]
+      yearExtrapolate <- -1
     }
-    
-    interpolated_year_int       <- as.integer(substring(single_interpolated_year,2))
-    year_before_int             <- as.integer(substring(year_before,2))
-    year_after_int              <- as.integer(substring(year_after,2))
-    
-    dataset_difference          <-  dataset[,year_after,,drop=FALSE] - dataset[,year_before,,drop=FALSE]
-    year_before_to_after        <-  year_after_int        - year_before_int
-    year_before_to_interpolated <-  interpolated_year_int - year_before_int
-    
-    
-    if(year_extrapolate== -1){
-      dataset_interpolated[,single_interpolated_year,] <- dataset[,year_before,,drop=FALSE]+ year_before_to_interpolated * dataset_difference / year_before_to_after
+
+    interpolatedYearInt       <- as.integer(substring(single_interpolated_year, 2))
+    yearBeforeInt             <- as.integer(substring(yearBefore, 2))
+    yearAfterInt              <- as.integer(substring(yearAfter, 2))
+
+    datasetDifference        <-  dataset[, yearAfter, , drop = FALSE] - dataset[, yearBefore, , drop = FALSE]
+    yearBeforeToAfter        <-  yearAfterInt        - yearBeforeInt
+    yearBeforeToInterpolated <-  interpolatedYearInt - yearBeforeInt
+
+
+    if (yearExtrapolate == -1) {
+      datasetInterpolated[, single_interpolated_year, ] <- dataset[, yearBefore, , drop = FALSE] +
+        yearBeforeToInterpolated * datasetDifference /
+          yearBeforeToAfter
     } else {
-      dataset_interpolated[,single_interpolated_year,] <- dataset[,year_extrapolate,,drop=FALSE]
+      datasetInterpolated[, single_interpolated_year, ] <- dataset[, yearExtrapolate, , drop = FALSE]
     }
   }
-  if(integrate_interpolated_years==FALSE) {
-    add_years <- setdiff(interpolated_year,interpolated_year_filtered)
-    if(length(add_years)>0){
-      dataset <- abind::abind(dataset_interpolated,dataset[,add_years,,drop=FALSE],along=2)
+  if (integrate_interpolated_years == FALSE) {
+    addYears <- setdiff(interpolated_year, interpolatedYearFiltered)
+    if (length(addYears) > 0) {
+      dataset <- abind::abind(datasetInterpolated, dataset[, addYears, , drop = FALSE], along = 2)
```

Comment:
`@importFrom abind abind` is missing here as well

Reply:
technically, it is sufficient to have one importFrom statement somewhere in the code, but I agree that it is cleaner to have it in every script which relies on it. I will add it